src/logical/execution/planning/plan_head_restricted.rs

```diff
@@ -149,10 +149,16 @@ impl HeadStrategy for RestrictedChaseStrategy {
 
         // 1. Compute the table "New Satisfied Matches"
 
+        let step_last_applied = if rule_info.step_last_applied == 0 {
+            0
+        } else {
+            rule_info.step_last_applied + 1
```

Comment:
Can you leave a comment on why it is necessary to add `1` here? At least to me this is not obvious :D

Reply:
I thinks it's not super clear to me yet but having a comment should certainly help in the future (to at least raise awareness that this part is non-trivial).